x/clp/client/cli/spam.go

```diff
@@ -0,0 +1,346 @@
+	"github.com/cosmos/cosmos-sdk/crypto/keys"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/wire"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	authcmd "github.com/cosmos/cosmos-sdk/x/auth/client/cli"
+	"github.com/cosmos/cosmos-sdk/x/bank/client"
+	tmcrypto "github.com/tendermint/tendermint/crypto"
+)
+
+// create new clp transaction
+func RunSpamCmd(cdc *wire.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "run_spammers <start_index>",
+		Short: "Run 10 previously created spammers starting from account index",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			fmt.Println("starting")
+			startIndex, _ := strconv.Atoi(args[0])
+			numSpammers := 80
+
+			runtime.GOMAXPROCS(runtime.NumCPU())
+
+			fmt.Println("Getting keybase")
+			kb, err := cosmosClientKeys.GetKeyBase() //XXX
+			if err != nil {
+				fmt.Println(err)
+				return err
+			}
+			fmt.Println("Got keybase")
+			fmt.Printf("Start Index is: %v\n", startIndex)
+
+			var wg sync.WaitGroup
+			wg.Add(numSpammers)
+
+			var spammers []Spammer
+			for i := startIndex; i < startIndex+numSpammers; i++ {
+				fmt.Printf("Spammer %v: Spawning...\n", i)
+				var buffer bytes.Buffer
+				buffer.WriteString("spammer")
+				buffer.WriteString(strconv.FormatInt(int64(i), 10))
+				accountName := buffer.String()
+
+				newSpammer := SpawnSpammer(accountName, i, kb)
+				spammers = append(spammers, newSpammer)
+				fmt.Printf("Spammer %v: Spawned...\n", i)
+			}
+
+			for i := startIndex; i < startIndex+numSpammers; i++ {
+				fmt.Printf("Spammer %v: Starting up...\n", i)
+				go spammers[i].start()
+				fmt.Printf("Spammer %v: Started...\n", i)
+			}
+
+			wg.Wait()
+
+			return nil
+		},
+	}
+}
+
+//yezs
+func CreateSpamCmd(cdc *wire.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "create_spammers <start_index>",
+		Short: "Create 20 spammers starting from account index",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			fmt.Println("starting")
+			password := "1234567890"
+			startIndex, err := strconv.Atoi(args[0])
+
+			richAccountName := "local_validator"
+
+			kb, err := cosmosClientKeys.GetKeyBase() //XXX
+			if err != nil {
+				fmt.Printf("Keybase get fail...\n")
+			}
+
+			ctx := context.NewCoreContextFromViper().WithDecoder(authcmd.GetAccountDecoder(cdc))
+
+			richFrom, err := GetFromAddress(kb, richAccountName)
+			if err != nil {
+				fmt.Println(err)
+				return err
+			}
+
+			ctx, err = SetupContext(ctx, richFrom)
+			if err != nil {
+				fmt.Println(err)
+				return err
+			}
+
+			richSequence, err := ctx.NextSequence(richFrom)
+			if err != nil {
+				fmt.Println(err)
+			}
+
+			for i := startIndex; i < startIndex+20; i++ {
+				fmt.Printf("Spammer %v: Creating Account...\n", i)
+				var buffer bytes.Buffer
+				buffer.WriteString("spammer")
+				buffer.WriteString(strconv.FormatInt(int64(i), 10))
+				accountName := buffer.String()
```

Comment:
Just out of curiosity: Is there a reason why you not just use `Sprintf("spammer-%v", i)` here?